### PR TITLE
OSHMEM: Let `pshmem.h` include `shmem.h` to be stand-alone again

### PR DESCRIPTION
**Pull request by bertwesarg**
See #6093

Signed-off-by: Bert Wesarg <bert.wesarg@tu-dresden.de>

### CODE REVIEW

**Comment by bertwesarg** [comment]
@jsquyres Why is this only an 'enhancement'? I consider this a regression re Open MPI <4.0.0 and thus should also be backported to 4.0.x.

**Comment by jsquyres** [comment]
Sure.  I have no real opinion here.

**Comment by jsquyres** [comment]
@yosefe @jladd-mlnx Ping.

**Comment by hoopoepg** [comment]
Fine for me too
Thank you

**Comment by bertwesarg** [comment]
What blocks this to be merged into master and than into v4.x?

**Comment by jsquyres** [comment]
@hoopoepg @yosefe You guys might not recognize that Bert is just a contributor and does not have merge permissions on the OMPI repos.  If you guys are ok with this PR (and you approved it, so I assume you are...?  😉), you should feel free to merge it.

File: oshmem/include/pshmem.h [hunk 1]
```diff
@@ -14,32 +14,7 @@
...
-#ifndef OSHMEM_DECLSPEC
-#  if defined(OPAL_C_HAVE_VISIBILITY) && (OPAL_C_HAVE_VISIBILITY == 1)
-#     define OSHMEM_DECLSPEC __attribute__((visibility("default")))
-#  else
-#     define OSHMEM_DECLSPEC
-#  endif
-#endif
-
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)
-#define OSHMEMP_HAVE_C11 1
```
Comment (ggouaillardet) [inline oshmem/include/pshmem.h]: if we remove this block, then we have to use the `OSHMEM_HAVE_C11` (note there is no `P`) macro (that is defined in `shmem.h`). @yosefe is that OK for you ?

Comment (yosefe) [reply oshmem/include/pshmem.h]: nice catch :+1:

Comment (bertwesarg) [reply oshmem/include/pshmem.h]: @yosefe do you take care of it or should I file a new pr?

Comment (ggouaillardet) [reply oshmem/include/pshmem.h]: @bertwesarg I just filed #6183

**Pull request merged**